components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/AssetBlob.java

```diff
@@ -35,6 +36,8 @@
 {
   private final BlobRef blobRef;
 
+  private final Blob blob;
```

Comment:
BlobRef contains indirection information too, where Blob is stored. BlobId is unique across given BlobStore, while BlobRef is unique across all (future) nodes, BlobRef is `store@node@blobId`. OTOH, BlobRef is used by Asset only, to address it's underlying Blob, so theoretically we can have locally an asset that would fetch something from "remote" node, and similarly, if Blobs get replicated, they would remain unchanged (merely just copied), but the target Asset would relink with different BlobRef... (this part is not yet fully worked out).

@jdillon any hints for Stuart's concerns?